### PR TITLE
Add integrated tests

### LINKED ISSUE #34: Create a better way to run tests
A Discord bot is really hard to mock so we should set up a proper testing server where MeowBot can be tested properly.

### PR DESCRIPTION
**Pull request by lwise**
First part of #34 

1. Adds a TesterBot which can currently send and read last message.
2. Converts `MessageListener` test to `MeowListenerTest`, since it makes more sense as an integration test.
3. Adds an `integrationTest` task and source set, which should be different from the `test` task/source set.
4. Updates the Github actions workflow to run the new integration test task with MeowBot. I also had to add in secrets to the repo's settings so we could grab them as environment variables.

This should theoretically work on CI, but I'll have to see when I open the pull request :p

### CODE REVIEW

File: src/main/kotlin/com/lwise/MeowBot.kt [hunk 1]
```diff
@@ -17,27 +17,33 @@ import discord4j.core.DiscordClientBuilder
...
 
 fun main() {
-    val dotenv = dotenv {
-        ignoreIfMissing = true
-    }
-    val client = DiscordClientBuilder.create(dotenv["BOT_TOKEN"] ?: System.getenv("BOT_TOKEN"))
-        .build()
-        .login()
-        .block()
+    MeowBot().runBot()
```
Comment (lwise) [inline src/main/kotlin/com/lwise/MeowBot.kt]: one sec gonna undo this change because we don't need it

**Review by shellychw — APPROVED** [review]
lgtm!
